kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -238,19 +235,45 @@
     private final AtomicLong pendingBytes = new AtomicLong(0);
     private volatile boolean autoReadDisabledPublishBufferLimiting = false;
 
+    private String getCurrentTenant() {
+        if (kafkaConfig.isKafkaEnableMultitenantMetadata()
+                && authenticator != null
+                && authenticator.session() != null
+                && authenticator.session().getPrincipal() != null) {
+            String username =  authenticator.session().getPrincipal().getTenantSpec();
+            if (username != null && !username.isEmpty()) {
+                // username can be "tenant" or "tenant/namespace"
+                if (username.contains("/")) {
+                    username = username.substring(0, username.indexOf('/'));
+                }
+                log.debug("using {} as tenant", username);
+                return username;
+            }
```

Comment:
Need rename the username to tenantSpec too.

Reply:
sure! good catch